File: 04-Arrays-Part-IV/python/05_count_subarrays_with_xor_k.py
class Solution:
    def subarraysWithXorK(self, nums: List[int], k: int) -> int:
        map = {0: 1}
        xor = 0
        count = 0

        for num in nums:
            xor ^= num
            count += map.get(xor ^ k, 0)
            map[xor] = map.get(xor, 0) + 1

        return count

File: 04-Arrays-Part-IV/java/05-CountSubarraysWithXorK.java
class Solution {
    public int subarraysWithXorK(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();

        int xor = 0;
        int count = 0;

        map.put(0, 1);

        for (int num : nums) {
            xor ^= num;

            count += map.getOrDefault(xor ^ k, 0);

            map.put(xor, map.getOrDefault(xor, 0) + 1);
        }

        return count;
    }
}
